uvls/src/main.rs

```diff
@@ -544,11 +557,60 @@ impl LanguageServer for Backend {
                         arguments: Some(vec![uri_json]),
                     }),
                     data: None,
-                }
+                },
             ]))
         }
     }
 
+    async fn code_action(&self, params: CodeActionParams) -> Result<Option<CodeActionResponse>> {
+        for diagnostic in params.context.diagnostics {
+            // Checks if there is a quick fix for the current diagnostic message
+            match diagnostic.message.as_str() {
+                "name contains a dash (-)" => {
+                    if let Ok(Some((Draft::UVL { source, .. }, ..))) =
+                        self.snapshot(&params.text_document.uri, false).await
+                    {
+                        let start_byte = byte_offset(&diagnostic.range.start, &source);
```

Comment:
maybe we should put this in a separate function (and even in a file like `uvl/core/action.rs` to make it more easy to extend with further code actions that might come in the future, this might blow up this `code_action` function a lot.